ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go

```diff
@@ -264,6 +297,84 @@ func generateNetworkBlackHolePortTestCases(name string, expectedHappyResponseBod
 				}, nil)
 			},
 		},
+		{
+			name:               fmt.Sprintf("%s no task network namespace", name),
+			expectedStatusCode: 500,
+			requestBody:        happyBlackHolePortReqBody,
+			expectedResponseBody: types.NewNetworkFaultInjectionErrorResponse(
+				fmt.Sprintf("failed to get task metadata due to internal server error for container: %s", endpointId)),
+			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
+				agentState.EXPECT().GetTaskMetadata(endpointId).Return(state.TaskResponse{
+					TaskResponse:          &v2.TaskResponse{TaskARN: taskARN},
+					FaultInjectionEnabled: true,
+					TaskNetworkConfig: &state.TaskNetworkConfig{
+						NetworkMode:       awsvpcNetworkMode,
+						NetworkNamespaces: nil,
+					},
+				}, nil)
```

Comment:
nit (non-blocking): might be a good idea to add a `Times(1)` for these test cases as well if want to make things consisent. wdyt?

Reply:
Sounds good. I can add it in a follow up change if no major concern in this PR.